addons/stock/models/stock_inventory.py

```diff
@@ -405,18 +370,18 @@ def write(self,vals):
 
     def _check_no_duplicate_line(self):
         for line in self:
-            existings = self.search([
+            domain = [
                 ('id', '!=', line.id),
                 ('product_id', '=', line.product_id.id),
-                ('inventory_id.state', '=', 'confirm'),
                 ('location_id', '=', line.location_id.id),
                 ('partner_id', '=', line.partner_id.id),
                 ('package_id', '=', line.package_id.id),
-                ('prod_lot_id', '=', line.prod_lot_id.id)])
+                ('prod_lot_id', '=', line.prod_lot_id.id),
+                ('inventory_id', '=', line.inventory_id.id)]
+            existings = self.search_count(domain)
             if existings:
-                raise UserError(_("You cannot have two inventory adjustments in state 'In Progress' with the same product (%s),"
-                                   " same location, same package, same owner and same lot. Please first validate"
-                                   " the first inventory adjustment before creating another one.") % (line.product_id.display_name))
+                raise UserError(_("There is already one inventory adjustment line for this product,"
+                                  " you should rather modify this one instead of creating a new one."))
```

Comment:
Not sure it's really useful since we check only inventory lines of the same inventory (before we checked in all active inventory). If it raises this error, the user has already all the information as the error raises just after he tried to save it (the line is still present in edit mode, with all fields filled).